pkg/data/file.go

```diff
@@ -56,7 +66,55 @@ func convertURLToBytes(url string) (b []byte, contentType string, filename strin
 	if strings.HasPrefix(url, "data:") {
 		return convertDataURIToBytes(url)
 	}
+	if matches := minioURLPattern.FindStringSubmatch(url); matches != nil {
+		if len(matches) < 2 {
+			return nil, "", "", fmt.Errorf("invalid blob storage url: %s", url)
+		}
+		return fetchFileFromBlobStorage(uuid.FromStringOrNil(matches[1]))
+	}
+	return fetchExternalURL(url)
+}
+
+func fetchFileFromBlobStorage(urlUID uuid.UUID) (b []byte, contentType string, filename string, err error) {
+	ctx := context.Background()
+	logger, _ := logger.GetZapLogger(ctx)
+	clients, err := client.GetClients(ctx, logger)
+	if err != nil {
+		return nil, "", "", err
+	}
+	defer clients.GRPCConn.Close()
```

Comment:
Relying on the code using `client.GetClients` to remember about closing `GRPCConn` (which isn't used for anything else) is rather optimistic. Imagine some package just needs the `client.BlobStorageClient`, it's likely the dev won't even consider closing this.

Reply:
I exposed the Close func for the Clients.